### ISSUE TITLE
Checkboxes are not grouped semantically for accessibility

### ISSUE THREAD

**Issue by dcplebranch** [solspace/craft3-freeform#2423]
### What happened?

When I create a checkbox field, the code that Freeform writes does not meet accessibility standards (WCAG Level A Info and Relationships: https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships)

For example, it wrote this code:

```
<div class="freeform-row">
    <div class="freeform-column-12 freeform-fieldtype-checkboxes freeform-column" data-field-container="preferredDays" data-field-type="checkboxes">
        <label class="freeform-label freeform-required" for="form-input-preferredDays">Preferred Days</label>
        <label for="form-input-preferredDays-0"><input type="checkbox" id="form-input-preferredDays-0" class="freeform-input" name="preferredDays[]" value="Mondays"  data-required>Mondays</label>
        <label for="form-input-preferredDays-1"><input type="checkbox" id="form-input-preferredDays-1" class="freeform-input" name="preferredDays[]" value="Tuesdays"  data-required>Tuesdays</label>
        <label for="form-input-preferredDays-2"><input type="checkbox" id="form-input-preferredDays-2" class="freeform-input" name="preferredDays[]" value="Wednesdays"  data-required>Wednesdays</label>
        <label for="form-input-preferredDays-3"><input type="checkbox" id="form-input-preferredDays-3" class="freeform-input" name="preferredDays[]" value="Thursdays"  data-required>Thursdays</label>
        <label for="form-input-preferredDays-4"><input type="checkbox" id="form-input-preferredDays-4" class="freeform-input" name="preferredDays[]" value="Fridays"  data-required>Fridays</label>
        <label for="form-input-preferredDays-5"><input type="checkbox" id="form-input-preferredDays-5" class="freeform-input" name="preferredDays[]" value="Saturdays"  data-required>Saturdays</label>
    </div>
</div>
```

The problem is that it's using `<label for="form-input-preferredDays"> `

But there is no input with that id. The label is trying to label a group, not a single checkbox. Screen readers will not correctly associate “Preferred Days” with the set of checkboxes.  This means users navigating by form fields won’t hear: Preferred Days, group of 6 checkboxes. Instead, they’ll just hear standalone checkboxes.

More info on ideal code for accessibility here: https://www.w3.org/WAI/tutorials/forms/grouping/

### Errors and Stack Trace (if available)

```shell

```

### How can we reproduce this?

1. Create a form
2. Choose checkboxes and drag it onto the form
3. Enter in a label "Testing"
4. Choose source: custom and type "one" "two" "three" etc.
5. Publish the form and view it on a page using a screenreader

### Freeform Edition

Pro

### Freeform Version

5.14.15

### Craft Version

4.17.4

### When did this issue start?

- [x] Unsure
- [ ] Fresh install of Freeform
- [ ] After upgrading from older Freeform version
- [ ] After upgrading from older Craft version
- [ ] After a change to hosting configuration
